lab_1/main.py

```diff
@@ -4,56 +4,168 @@
 """
 
 
-def tokenize(text: str) -> list:
-    """
+#data = open('data.txt', encoding='utf-8').read()
+data = 'The weather is sunny, the man is happy. man man happy'
+def tokenize(text):
+    if type(text) != str:
+        return []
+    else:
+        lines = text.split('\n')
+        tokens = []
+        for words in lines:
+            words = words.split ()
+            for word in words:
+                word = word.lower()
+                word_new = ''
+                for letter in word:
+                    if letter.isalnum():# or letter in "'-/"
+                        word_new += letter
+                if word_new != '':
+                    tokens.append(word_new)
+        return tokens
+
+tokens = tokenize (data) 
+"""
     Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
     :param text: the initial text
     :return: a list of lowercased tokens without punctuation
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    """
-    return 0
+"""
 
 
-def remove_stop_words(tokens: list, stop_words: list) -> list:
-    """
+stop_words = open('stop_words.txt', encoding='utf-8').read()
```

Comment:
you shouldn't open anything in main.py.
"stop_words" is an argument of the remove_stop_words function
a user will add it, you just work with it considering that it already exists